### PR TITLE
Removing binary dependency

### LINKED ISSUE #1: Remove unnecessary binary package dependency for running on node
The 'binary', and by extension 'buffers', packages are old dependencies that no longer have corresponding codebases, and the 'buffers' package has no license associated with it.  It looks like the functionality provided by these dependencies now exists in node's built-in modules.   Remove the dependency on the 'binary' package to instead use node's built-in buffer functionality.

Note: This change will be an issue for cross-browser compatibility.

### PR DESCRIPTION
**Pull request by amcgoogan**
Removing dependency on the 'binary' package and using node's built-in buffer module to perform unsigned LE number parsing.

testing:
npm install
npm test

### CODE REVIEW

File: lib/parseBuffer.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+ *   ...
+ * ]
+ * @returns An object with keys set to their associated extracted values.
+ */
+const parse = function(buffer, format) {
+  var result = {}
+  var offset = 0;
+  for(const [key, size] of format) {
+    if(buffer.length >= offset + size) {
+      result[key] = parseUIntLE(buffer, offset, size);
```
Comment (pdugas) [inline lib/parseBuffer.js:43]: Should this be `bufer.parseUIntLE()` so it's consistent with they way it's used above?

File: lib/parseBuffer.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+ * [
+ *   [
+ *     <key>,  // Name of the key to assign the extracted number to.
+ *     <size>  // The size in bytes of the number to extract. possible values are 1, 2, 4, 8.
+ *   ],
+ *   ...
+ * ]
+ * @returns An object with keys set to their associated extracted values.
+ */
+const parse = function(buffer, format) {
```
Comment (pdugas) [inline lib/parseBuffer.js:38]: Should we UT this function?

**Review by pdugas — CHANGES_REQUESTED** [review]
Checked the byte lengths of the translated `parse()` calls and don't see any diffs.  

Had one nit you can ignore if you like. 

Think we need to add a UT for the new `parseBuffer()` function.

File: lib/parseBuffer.js (hunk 2 above)
Comment (amcgoogan) [reply lib/parseBuffer.js:38]: Added test.

File: lib/parseBuffer.js (hunk 1 above)
Comment (pdugas) [reply lib/parseBuffer.js:43]: duh, being an idiot...

**Review by pdugas — APPROVED** [review]

**Review by zach-cribl — APPROVED** [review]
Removes old package with problematic license and maintains needed functionality 👍

**Review by nromito — APPROVED** [review]

**Pull request merged**